yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -163,13 +180,13 @@ export class PeerManager implements PeerManagerInterface {
    */
   private handleConnectedPeerEvent(e: CustomEvent<PeerId>) {
     const peerId = e.detail;
-    if (this.peerDiscoveryService.isBootstrapPeer(peerId)) {
-      this.logger.verbose(`Connected to bootstrap peer ${peerId.toString()}`);
-    } else {
-      this.logger.verbose(`Connected to transaction peer ${peerId.toString()}`);
+    if (this.config.p2pDisableStatusHandshake) {
+      return;
```

Comment:
Let's add a check during initialization that throws if `p2pDisableStatusHandshake` and `p2pAllowOnlyValidators` are both set to true.